updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,222 @@
+
+However there might be some secenarios where a controlled rollout of upgrade to the worker nodes is desired.
+This helps to make sure the mission critical application stays available during the whole update process even if the upgrade process causes a failure.
+It is also useful in scenarios where we can not have a single maintenance window to complete the whole update process.
+
+The slow rollout of a new OpenShift version to worker nodes can be characterized as a canary release i.e. we can
+control the rollout of worker nodes by controlling the machine config pools. After the update of the first MCP, the
+application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+
+== Workflow for updating worker nodes with canary rollout
+
+. Create MCPs out of the worker pool. The number of nodes in each MCP depends on a few factors like maintaince window duration for each MCP and the amount of reserve capacity (extra worker nodes) present.
++
+[NOTE]
+====
+In case of a failure i.e. if the MCP with the new version does not work as expected with the applications, the nodes in the pool can be cordoned and drained. So that the application will not run on these nodes and the extra capacity will help to maintain the quality of service of the applications.
+====
++
+. Pause the MCPs you do not want to update as part of the default update process.
++
+[NOTE]
+====
+Pausing the MCP will also pause the kube-apiserver-to-kubelet-signer automatic CA certificates rotation.
+New CA certificates are generation and removal happens at 292 day (from the installation day) and 365 days respectively.
+The renewal and removal of the certs need reboot of the nodes.
+If the MCPs are paused then the node reboots will not happen and that would make the cluster degraded.
+See the link:https://access.redhat.com/articles/5651701[article] to findout how much time you have before the next automatic CA certificate rotation.
+Make sure the pools are unpaused when the CA cert rotation happens.
+====
++
+. Start the update process. The update process will only update the MCPs which are not paused. That includes the master nodes i.e. the control plane.
++
+. Once the control plane update is completed, unpause one MCP. Unpausing the MCP will start the update process for the pool of nodes. You can check the progress of the update in the web console (In Administrator view -> Administration -> Cluster settings ) as well as running `$ oc get machineconfigpools` CLI command.
++
+[NOTE]
+====
+You can change `maxUnavailable` in a MCP to specify the percentage or the number of machines that can be updating at any given time. The default is 1.
+====
++
+. Test if the applications are working as expected on the newly updated MCP.
+
+. Update the remaining MCPs By unpausing one by one till all worker nodes are updated.
+
+. Unpause the MCPs you had paused in the previous step.
```

Comment:
Sounds like this sentence is a bit redundant based on the above sentence is clear enough.

Reply:
You are right. Will fix it.